be/src/connector/hive_connector.cpp

```diff
@@ -744,6 +744,14 @@ Status HiveDataSource::_init_scanner(RuntimeState* state) {
     } else if (use_kudu_jni_reader) {
         scanner = create_kudu_jni_scanner(jni_scanner_create_options).release();
     } else if (format == THdfsFileFormat::PARQUET) {
+        scanner_params.parquet_page_index_enable =
+                config::parquet_page_index_enable
+                        ? state->query_options().enable_parquet_reader_page_index ? true : false
```

Comment:
Should check __isset.enable_parquet_reader_page_index first?

state->query_options().enable_parquet_reader_page_index ? true : false  => state->query_options().enable_parquet_reader_page_index?